backends/utils/history_server.go

```diff
@@ -0,0 +1,617 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/nuclio/logger"
+	"github.com/pkg/errors"
+	"github.com/v3io/frames"
+	"github.com/v3io/frames/pb"
+	"github.com/v3io/frames/v3ioutils"
+	v3io "github.com/v3io/v3io-go/pkg/dataplane"
+	v3iohttp "github.com/v3io/v3io-go/pkg/dataplane/http"
+	v3ioerrors "github.com/v3io/v3io-go/pkg/errors"
+	"github.com/v3io/v3io-tsdb/pkg/utils"
+)
+
+const (
+	defaultHistoryFileDurationInSeconds = 24 * 3600
+	defaultHistoryFileNum               = 7
+	defaultWriteMonitoringLogsTimeout   = 10 * time.Second
+	defaultPendingLogsBatchSize         = 100
+	defaultLogsFolderPath               = "/monitoring/frames"
+	defaultLogsContainer                = "users"
+	defaultMaxBytesInNginxRequest       = 5 * 1024 * 1024 // 5MB
+	maxLogsInMessage                    = 1000
+	maxRetryNum                         = 5
+
+	queryType   = "query"
+	ingestType  = "ingest"
+	createType  = "create"
+	deleteType  = "delete"
+	executeType = "execute"
+
+	logFileTimeFormat = "2006-01-02T15"
```

Comment:
You're missing four digits and a Z at the end :)

Reply:
💩